app/Config.hs

```diff
@@ -82,7 +83,13 @@ instance FromJSON Config where
 
 -- Load the configuration
 loadConfig :: IO Config
-loadConfig = loadYamlSettings ["config.yaml"] [] useEnv
+loadConfig = do
+    env <- lookupEnv "APP_ENV"
+    let configFile = case env of
+            Just "test" -> "test.config.yaml"
```

Comment:
This is on the right track, but can still be improved further by taking advantage of the fact that `loadYamlSettings` takes a list: when the environment is `test`, load *both* `test.config.yaml` and `config.yaml`. That way `test.config.yaml` only has to override the defaults provided in `config.yaml`.